src/main/scala/dotty/dokka/ScalaSignatureProvider.scala

```diff
@@ -0,0 +1,106 @@
+            builder.generics(clazz)
+            ext.constructor.foreach(c => builder.functionParameters(c))
+            ext.parentTypes match 
+                case Nil =>
+                case extendType :: withTypes =>
+                    builder.addText(" extends ") 
+                    builder.typeSignature(extendType)
+                    withTypes.foreach { t => 
+                        builder.addText(" with ")  
+                        builder.typeSignature(extendType)
+                    }
+        }
+
+    private def methodSignature(method: DFunction): ContentNode = 
+        content(method){ builder =>
+            utils.annotationsBlock(builder, method)
+            // builder.addText("TODO modifiers")
+            builder.addText("def")
+            builder.addText(" ")
+            builder.addLink(method.getName, method.getDri)
+            builder.generics(method)  
+            builder.functionParameters(method)
+            builder.addText(":")
+            builder.addText(" ")
+            builder.typeSignature(method.getType)
+        }
+
+
+    extension on (builder: PageContentBuilder$DocumentableContentBuilder):
+        def typeSignature(b: Projection): Unit = b match {
+            case tc: TypeConstructor =>
+                tc.getProjections.asScala.foreach {
+                    case text: UnresolvedBound => builder.addText(text.getName)
+                    case link: OtherParameter => 
+                        builder.addLink(link.getName, link.getDeclarationDRI)
+                    case other =>
+                        builder.addText(s"TODO($other)")
+                }
+            case other =>
+             builder.addText(s"TODO: $other")
+        }
+
+        def generics(on: WithGenerics) = builder.addList(on.getGenerics, "[", "]"){ e => 
+            builder.addText(e.getName)
+            e.getBounds.forEach(b => builder.typeSignature(b))
+        }
+        
+        def functionParameters(method: DFunction) = 
+            val methodExtension = method.get(MethodExtension)
+            methodExtension.parametersListSizes.foldLeft(0){ (from, size) =>
+                val toIndex = from + size
+                if from == toIndex then builder.addText("()")
+                else builder.addList(method.getParameters.subList(from, toIndex), "(", ")"){ param =>
+                    utils.annotationsInline(builder, param)
+                    // builder.addText("TODO modifiers")
+                    builder.addText(param.getName)
+                    builder.addText(": ")
+                    builder.typeSignature(param.getType)
+                }
+                toIndex
+            }
+
+    private def content(d: Documentable)(render: PageContentBuilder$DocumentableContentBuilder => Unit): ContentGroup = 
+        val lambda: Consumer[PageContentBuilder$DocumentableContentBuilder] = new Consumer:
+          override def accept(v: PageContentBuilder$DocumentableContentBuilder) = render(v)
```

Comment:
It should be possible to write `v => render(v)` instead of `new Consumer { ... }`, lambdas are automatically "adapted" when the expected type is a SAM.